### PR TITLE
[Feat-650] Support update project info (#650)

### PR DESCRIPTION
**Pull request by lvshaokang**
backend:

* [feat] add /flink/project/get /flink/project/update api

* [improve] improve log format by adding line number

frontend:

* [feat] add edit view

### CODE REVIEW

**Comment by lvshaokang** [comment]
@wolfboys PTAL

**Comment by wolfboys** [comment]
> @wolfboys PTAL
good job, thank u for your hard work, I'll review it later

**Review by xinzhuxiansheng — COMMENTED** [review]
图片

The logback changes you mentioned can help developers quickly locate problems, but it is recommended that 'L' will have performance loss as described on the official website (ps : refer to https://logback.qos.ch/manual/layouts.html ), and refer to other open source project log log printing, almost never let go 'L'  .

图片

你提的logback改动帮助开发快速定位问题，但建议是根据官网描述的那样 'L'是会有性能损耗，以及参考其他开源项目log日志打印，几乎都不放开 'L' 。 

建议 不打印 L,  logback可以让用户自己定义即可

**Comment by xinzhuxiansheng** [comment]
图片

Avoid using e.printStackTrace(); to print log

**Comment by lvshaokang** [comment]
@xinzhuxiansheng thanks for your review, i indeed ignored performance, and i will change it.

**Comment by xinzhuxiansheng** [comment]
You have added a buildState state, -2, and I really want to discuss the difference between -1 and -2 with you. I feel that these two are semantically similar.   
Project.java
```java
    /**
     * 构建状态: -1:未构建 0:正在构建 1:构建成功 2:构建失败
     */
    @TableField("BUILDSTATE")
    private Integer buildState;
``` 

你新增了一个buildState状态，-2，非常想与你讨论 -1，-2的差异化，自己感觉这两个在语义上较相似  😊

**Comment by lvshaokang** [comment]
@xinzhuxiansheng I think these two states(not build & rebuild ) all need to trigger build operation,so i use negative number as rebuild state

File: streamx-console/streamx-console-service/src/main/java/com/streamxhub/streamx/console/core/controller/ProjectController.java [hunk 1]
```diff
@@ -54,6 +54,18 @@ public RestResponse create(Project project) {
...
+    @PostMapping("update")
+    @RequiresPermissions("project:update")
+    public RestResponse update(Project project) {
+        boolean update = projectService.update(project);
+        return RestResponse.create().data(update);
+    }
+
+    @PostMapping("get")
+    public RestResponse get(Long id) {
+        return RestResponse.create().data(projectService.getProject(id));
```
Comment (wolfboys) [inline streamx-console/streamx-console-service/src/main/java/com/streamxhub/streamx/console/core/controller/ProjectController.java:66]: projectService.getById(id)


there are already methods to query objects according to the primary key. There is no need to rewrite another method

**Review by wolfboys — COMMENTED** [review]
LGTM.

**Pull request merged**